cmd/thanos/query.go

```diff
@@ -61,9 +61,6 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 	webExternalPrefix := cmd.Flag("web.external-prefix", "Static prefix for all HTML links and redirect URLs in the UI query web interface. Actual endpoints are still served on / or the web.route-prefix. This allows thanos UI to be served behind a reverse proxy that strips a URL sub-path.").Default("").String()
 	webPrefixHeaderName := cmd.Flag("web.prefix-header", "Name of HTTP request header used for dynamic prefixing of UI links and redirects. This option is ignored if web.external-prefix argument is set. Security risk: enable this option only if a reverse proxy in front of thanos is resetting the header. The --web.prefix-header=X-Forwarded-Prefix option can be useful, for example, if Thanos UI is served via Traefik reverse proxy with PathPrefixStrip option enabled, which sends the stripped prefix value in X-Forwarded-Prefix header. This allows thanos UI to be served on a sub-path.").Default("").String()
 
-	queryTimeout := modelDuration(cmd.Flag("query.timeout", "Maximum time to process query by query node.").
```

Comment:
I would leave this as it was, reasons:

1. `query` sounds like being related to QueryAPI, that's why this name. Which always included `promql` one. That's why I would leave like it was, the fact that query also queries something else underneath is hidden. Also it matches Prometheus `query` flags as well. And yet another thing compatibility. Even though we are not `1.0` this API change is  major hit into compatibiltiy as suddently query timeout means something opposite to what it was ):

2. `promql` sounds like only promQL, however we know it involves `proxy StoreAPI Series()` invocation. More over not only one, but sometimes more then one! PromQL can run multiple `Select`s per single Query. 

I really like this idea, but IMO it should be:
`query.timeout` (as it was)
`store.timeout`  for proxy.go client timeout.

alternatively if we want to be more explicit maybe `query.select-timeout`? That will also include then the deduplication process.